homeassistant/components/heos/media_player.py

```diff
@@ -167,9 +169,49 @@ def __init__(self, player):
         """Play a piece of media."""
         if media_type == MEDIA_TYPE_URL:
             await self._player.play_url(media_id)
-        else:
-            _LOGGER.error("Unable to play media: Unsupported media type '%s'",
-                          media_type)
+            return
+
+        if media_type == "quick_select":
+            # media_id may be an int or a str
+            selects = await self._player.get_quick_selects()
+            try:
+                index = int(media_id)
+            except ValueError:
+                # Try finding index by name
+                index = next((index for index, select in selects.items()
+                              if select == media_id), None)
+            if index is None:
+                raise ValueError("Invalid quick select '{}'".format(media_id))
+            await self._player.play_quick_select(index)
+            return
+
+        if media_type == MEDIA_TYPE_PLAYLIST:
+            from pyheos import const
```

Comment:
You can import at the top :)

Reply:
I'll go through and update the entire integration at once.